syntax = "proto3";

service BankingService {
  rpc MsgDelivery(BankingRequest) returns (BankingResponse);
  rpc Propagate_Deposit (PropagateRequest) returns (BankingResponse);
  rpc Propagate_Withdraw (PropagateRequest) returns (BankingResponse);
}

message BankingRequest {
  string customer_request_id = 1;
  string id = 2;
  string interface = 3;
  int32 money = 4;
  int32 clock = 5;
}

message PropagateRequest {
  string customer_request_id = 1;
  string id = 2;
  int32 money = 3;
  int32 clock = 4;
}

message BankingResponse {
  string customer_request_id = 1;
  string customer_id = 2;
  string response_message = 3;
  int32 clock = 4;
}

message CustomerEvent {
  string customer_request_id = 1;
  string interface = 2;
  int32 money = 3;
  int32 clock = 4;
}